pkg/validation/validate_bundle.go

```diff
@@ -27,7 +27,21 @@ type BundleValidator struct {
 	Bundle *bpb.ClusterBundle
 }
 
-var apiVersionPattern = regexp.MustCompile(`^gke.io/k8s-cluster-bundle/\w+$`)
+var (
+	apiVersionPattern = regexp.MustCompile(`^gke.io/k8s-cluster-bundle/\w+$`)
+
+	// Regex string for numbers without leading zeros.
+	number = `([1-9]\d*|0)`
+	// Regex string for dot separated alphanumeric identifier with hyphen.
+	dotID  = `[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*`
+	// The three regex strings below make up the SemVer 2.0.0 string.
+	// MAJOR.MINOR.PATCH-preRelease+metaData
+	majorMinorPatch = fmt.Sprintf(`%s\.%s\.%s`, number, number, number)
+	preRelease      = fmt.Sprintf(`(-%v)?`, dotID)
```

Comment:
I wonder if we need the major-minor-patch junk to simplifying things. I think we might want it for the component API Version, but I think we might want to enforce a subset of Sem-Ver (Major-Minor-Patch).  WDYT? Do you think we need this?